source/MainPanel.cpp

```diff
@@ -109,8 +111,13 @@ void MainPanel::Step()
 		{
 			// Boarding events are only triggered by your flagship.
 			Mission *mission = player.BoardingMission(event.Target());
-			if(mission)
+			if(mission && mission->HasSpace(player))
```

Comment:
It would make more sense for boarding missions to never offer if they would make you jettison cargo.

Also note that the check actually needs to be done based on whether the player's flagship has space; Mission::HasSpace() is checking the CargoHold object in the player info.

Reply:
I've addressed the latter point in commit 1d78031, since it was a bug in the existing code.

Reply:
This PR updates Mission::HasSpace to check the appropriate CargoHold. However, by not allowing the missions to offer if any jettison, even of commodities, is required, I can drop the forced ShipInfoPanel bits.